cardano-cli/src/Cardano/CLI/Shelley/Commands.hs

```diff
@@ -348,6 +351,19 @@ renderPoolCmd cmd =
     PoolGetId {} -> "stake-pool id"
     PoolMetadataHash {} -> "stake-pool metadata-hash"
 
+data TxMempoolQuery =
+      TxMempoolQueryTxExists TxId
+    | TxMempoolQueryNextTx
+    | TxMempoolQueryInfo
+  deriving Show
+
+renderTxMempoolQuery :: TxMempoolQuery -> Text
+renderTxMempoolQuery query =
+  case query of
+    TxMempoolQueryTxExists tx -> "tx-exists " <> pack (show tx) -- TODO render this properly
```

Comment:
It's enough to render the `TxId` here instead of the entire tx. You should be able to use `serialiseToRawBytesHexText`.

Reply:
Done